### PR TITLE
Fix group-by fields and Timeframe type toggle (PLUS: lodash upgrade)

### PR DESCRIPTION
**Pull request by aroc**
## What does this PR do? How does it affect users?

It fixes two bugs in Explorer:

1) When adding a second group by option, you could not remove that second group by field after adding it. This was due to how updates are merged into the model, a change that was introduced in #98.
2) When trying to toggle between absolute and relative timeframes, it would not work. You'd get "stuck" on an absolute timeframe after selecting it. This was due to how updates are merged into the model, a change that was introduced in #98.

There are two new tests added to cover these two scenarios.

#### PLUS: Upgrade to lodash 4.0.0
This PR also upgrades to latest lodash because we needed the `_.mergeWith` function to fix these bugs. :palm_tree: 

## How should this be tested?

1) Try using the group by fields in the query builder. Ensure they work as expected.
2) Try toggling between absolute and relative timeframes. Ensure they work as expected.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [ ] Does the code compile without warnings (check shell, console)?
- [ ] Do all tests pass?
- [ ] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

### CODE REVIEW

File: client/js/app/stores/ExplorerStore.js [hunk 1]
```diff
@@ -128,7 +128,14 @@ function _getDefaultFilterCoercionType(explorer, filter) {
...
   // don't have to loop through the (sometimes) massive response object.
-  var newModel = _.merge({}, explorer, _.omit(updates, 'response'));
+  function customizer(objValue, srcValue, key, object, source) {
+    if (_.isArray(objValue)) {
+      return srcValue;
+    } else if (key === 'time' && _.isPlainObject(objValue)) {
+      return srcValue;
+    }
+  }
+  var newModel = _.mergeWith({}, explorer, _.omit(updates, 'response'), customizer);
```
Comment (joannecheng) [inline client/js/app/stores/ExplorerStore.js]: I'm getting an error that `mergeWith` is not a function with the latest version of lodash.

File: client/js/app/stores/ExplorerStore.js [hunk 2]
```diff
@@ -128,7 +128,14 @@ function _getDefaultFilterCoercionType(explorer, filter) {
 function _prepareUpdates(explorer, updates) {
   // TODO: We're assigning the response object directly onto the model so we
   // don't have to loop through the (sometimes) massive response object.
-  var newModel = _.merge({}, explorer, _.omit(updates, 'response'));
+  function customizer(objValue, srcValue, key, object, source) {
```
Comment (joannecheng) [inline client/js/app/stores/ExplorerStore.js]: `object` and `source` aren't used anywhere in the function. It might be a bit cleaner to remove them from the argument list.

File: test/unit/stores/ExplorerStoreSpec.js [hunk 3]
```diff
@@ -221,6 +221,62 @@ describe('stores/ExplorerStore', function() {
...
+      var explorer = ExplorerStore.get('SOME_ID');
+      var updates = { query: { group_by: ['name'] } }
+      ExplorerActions.update('SOME_ID', updates);
+
+      explorer = ExplorerStore.get('SOME_ID');
+      
+      assert.sameMembers(explorer.query.group_by, ['name']);
+    });
+
+    it('should properly the time object', function () {
```
Comment (joannecheng) [inline test/unit/stores/ExplorerStoreSpec.js]: should properly what the time object? :)

**Comment by joannecheng** [comment]
Just a few comments, but otherwise everything works as it should. :confetti_ball:

File: client/js/app/stores/ExplorerStore.js (hunk 2 above)
Comment (aroc) [inline client/js/app/stores/ExplorerStore.js]: Agreed!

File: test/unit/stores/ExplorerStoreSpec.js (hunk 3 above)
Comment (aroc) [inline test/unit/stores/ExplorerStoreSpec.js]: lewl! good catch

**Comment by aroc** [comment]
Cool! Fixed up. Merging.
